dask/dataframe/core.py

```diff
@@ -2214,18 +2216,22 @@ def mean(
             num = self._get_numeric_data()
             s = num.sum(skipna=skipna, split_every=split_every)
             n = num.count(split_every=split_every)
-            name = self._token_prefix + "mean-%s" % tokenize(self, axis, skipna)
-            result = map_partitions(
-                methods.mean_aggregate,
-                s,
-                n,
-                token=name,
-                meta=meta,
-                enforce_metadata=False,
-                parent_meta=self._meta,
-            )
-            if isinstance(self, DataFrame):
-                result.divisions = (self.columns.min(), self.columns.max())
+            # Starting in pandas 2.0, `axis=None` does a full aggregation across both axes
+            if axis is None and not isinstance(s, Scalar) and PANDAS_GT_200:
+                result = s.sum() / n.sum()
+            else:
+                name = self._token_prefix + "mean-%s" % tokenize(self, axis, skipna)
+                result = map_partitions(
+                    methods.mean_aggregate,
+                    s,
+                    n,
+                    token=name,
+                    meta=meta,
+                    enforce_metadata=False,
+                    parent_meta=self._meta,
+                )
+                if isinstance(result, DataFrame):
```

Comment:
In `_reduction_agg`, you do this for both `DataFrame` and `Series`, but here just `DataFrame`. Is this enough?

Reply:
It should be okay as the `axis=None` case (where a `Scalar` is returned, not a `Series`) is handled in the `if`-block above. `_reduction_agg` is handling both cases, so needs to be a bit more flexible. That said, I included a commit that should make the logic (hopefully) easier to reason about